gpu-enabler/src/main/scala/com/ibm/gpuenabler/CUDADSUtils.scala

```diff
@@ -397,14 +436,16 @@ object CUDADSImplicits {
         case _ => ds.queryExecution.optimizedPlan
       }
       GPUSparkEnv.get.gpuMemoryManager.cacheGPUSlaves(md5HashObj(logPlan))
-
-      // Create a new Dataset to load the data into GPU
-      val ds1 = DS[T](ds.sparkSession,
-        LOADGPU[T](getPartSizes, getLogicalPlan(ds)))
-
-      // trigger an action
-      ds1.count()
-      ds1
+      if(GPUSparkEnv.get.isGPUEnabled) {
```

Comment:
No need to handle 'isGPUEnabled' in driver side of the code. This is should be handled at the physical plan's 'execute' method.